### PR TITLE
dependencies

### PR DESCRIPTION
**Pull request by Joeysantoro**

### CODE REVIEW

File: contract-addresses/dependencies.ts [hunk 1]
```diff
@@ -1,5 +1,564 @@
...
+    contractDependencies: []
+  },
+  optimisticTimelock: {
+    contractDependencies: ['core', 'rewardsDistributorAdmin', 'tribalChiefSync', 'rariPool8Comptroller']
+  },
+  rariTimelock: {
+    contractDependencies: []
+  },
+  tribeRariDAO: {
+    contractDependencies: []
```
Comment (ElliotFriedman) [inline contract-addresses/dependencies.ts:146]: isn't this dependent on the tribe token?

File: contract-addresses/dependencies.ts [hunk 2]
```diff
@@ -1,5 +1,564 @@
...
+    contractDependencies: ['core']
+  },
+  daiBondingCurve: {
+    contractDependencies: ['core']
+  },
+  daiPCVDripController: {
+    contractDependencies: ['core']
+  },
+  daiPSM: {
+    contractDependencies: ['core']
```
Comment (ElliotFriedman) [inline contract-addresses/dependencies.ts:164]: this is also dependent on the compoundPCVDripController

File: contract-addresses/dependencies.ts [hunk 3]
```diff
@@ -1,5 +1,564 @@
...
+  },
+  ethReserveStabilizer: {
+    contractDependencies: ['core']
+  },
+  tribeReserveStabilizer: {
+    contractDependencies: ['core']
```
Comment (ElliotFriedman) [inline contract-addresses/dependencies.ts:170]: wouldn't both reserve stabilizers be dependent on their respective drippers?

File: contract-addresses/dependencies.ts [hunk 4]
```diff
@@ -1,5 +1,735 @@
...
+    contractDependencies: ['core', 'fei', 'aaveEthPCVDeposit', 'ethReserveStabilizer']
+  },
+  bondingCurve: {
+    contractDependencies: ['core', 'fei', 'aaveEthPCVDeposit', 'compoundEthPCVDeposit', 'chainlinkEthUsdOracleWrapper']
+  },
+  compoundEthPCVDripController: {
+    contractDependencies: ['core', 'fei', 'compoundEthPCVDeposit', 'ethReserveStabilizer']
+  },
+  daiBondingCurve: {
+    contractDependencies: ['core', 'fei', 'compoundDaiPCVDeposit', 'chainlinkDaiUsdOracleWrapper']
```
Comment (ElliotFriedman) [inline contract-addresses/dependencies.ts:231]: because this is deprecated, can't we just remove?

File: contract-addresses/dependencies.ts [hunk 5]
```diff
@@ -1,5 +1,735 @@
...
+    contractDependencies: ['core', 'tribeUsdCompositeOracle', 'tribeMinter', 'collateralizationOracleWrapper', 'tribe']
+  },
+  aaveEthPCVDeposit: {
+    contractDependencies: ['core', 'aaveEthPCVDripController', 'bondingCurve', 'pcvGuardian']
+  },
+  aaveFeiPCVDeposit: {
+    contractDependencies: ['core', 'fei']
+  },
+  aaveRaiPCVDeposit: {
+    contractDependencies: ['core']
```
Comment (ElliotFriedman) [inline contract-addresses/dependencies.ts:265]: this should also be dependent on the aRAI token

File: contract-addresses/dependencies.ts [hunk 6]
```diff
@@ -1,5 +1,735 @@
...
+    ]
+  },
+  tribeReserveStabilizer: {
+    contractDependencies: ['core', 'tribeUsdCompositeOracle', 'tribeMinter', 'collateralizationOracleWrapper', 'tribe']
+  },
+  aaveEthPCVDeposit: {
+    contractDependencies: ['core', 'aaveEthPCVDripController', 'bondingCurve', 'pcvGuardian']
+  },
+  aaveFeiPCVDeposit: {
+    contractDependencies: ['core', 'fei']
```
Comment (ElliotFriedman) [inline contract-addresses/dependencies.ts:262]: this should also be dependent on the aFEI token

File: contract-addresses/dependencies.ts [hunk 7]
```diff
@@ -1,5 +1,735 @@
...
+  },
+  rariPool79FeiPCVDeposit: {
+    contractDependencies: ['core', 'fei']
+  },
+  rariPool7FeiPCVDeposit: {
+    contractDependencies: ['core', 'fei']
+  },
+  rariPool7LusdPCVDeposit: {
+    contractDependencies: ['core']
```
Comment (ElliotFriedman) [inline contract-addresses/dependencies.ts:358]: missing lusd, and the corresponding fToken

**Comment by ElliotFriedman** [comment]
It seems that this PR does not factor into account external dependencies which I would think it should.

File: contract-addresses/dependencies.ts (hunk 5 above)
Comment (Joeysantoro) [reply contract-addresses/dependencies.ts:265]: aRAI is external

File: contract-addresses/dependencies.ts (hunk 4 above)
Comment (Joeysantoro) [reply contract-addresses/dependencies.ts:231]: not deprecated yet

File: contract-addresses/dependencies.ts (hunk 6 above)
Comment (Joeysantoro) [reply contract-addresses/dependencies.ts:262]: aFEI is external

File: contract-addresses/dependencies.ts (hunk 7 above)
Comment (Joeysantoro) [reply contract-addresses/dependencies.ts:358]: both external

**Comment by Joeysantoro** [comment]
> It seems that this PR does not factor into account external dependencies which I would think it should.

disagree. Maybe we can add external later but this PR is already unwieldy. The point is to keep a mental map of internal contract dependencies primarily. The extra effort to add external deps isn't really worth it. I think it should be merged as-is.

**Review by ditchfieldcaleb — DISMISSED** [review]

**Pull request merged**